### PR TITLE
Enable repositories gradually during upgrade

### PR DESCRIPTION
**Pull request by lpramuk**
We are required to enable internal repositories gradually to perform exact steps in the same order described by the documentation.

Next version maintenance repository has to be enabled first, then is performed self-upgrade and after that we can enable other internal  repositories.

Coverage for [BZ#2143451](https://bugzilla.redhat.com/show_bug.cgi?id=2143451)

Depends on satelliteqe-jenkins `!995`

### CODE REVIEW

**Comment by lpramuk** [comment]
Tested on internal premises.
See  `SAT-14222` for details

File: upgrade/helpers/tasks.py [hunk 1]
```diff
@@ -876,46 +920,24 @@ def satellite_upgrade(zstream=False):
...
         )
-        # use Beta until becomes GA
-        if settings.upgrade.to_version == '6.12':
-            with shell_env(FOREMAN_MAINTAIN_USE_BETA='1'):
-                run(command)
-        else:
+        with (
+            shell_env(FOREMAN_MAINTAIN_USE_BETA='1')
+            if satellite and settings.upgrade.use_beta
+            else nullcontext()
```
Comment (devendra104) [inline upgrade/helpers/tasks.py:924]: nice addition

File: upgrade/helpers/tasks.py [hunk 2]
```diff
@@ -801,67 +801,111 @@ def check_ntpd():
...
+    else:
+        repository_setup(**CUSTOM_SAT_REPO['maintenance'])
+
+
+def foreman_maintain_self_upgrade(zstream=False, fetch_content_from_sat=False):
+    """
+    Install the latest fm rubygem-foreman_maintain to get the latest y-stream upgrade path.
+    """
+    if zstream:
+        run('foreman-maintain upgrade list-versions', warn_only=True)
```
Comment (devendra104) [inline upgrade/helpers/tasks.py:875]: Isn't it possible to have scenarios where we can receive updates in the maintenance package for the Z-stream?

Comment (devendra104) [reply upgrade/helpers/tasks.py:875]: I believe this command will check if any updates are available and, if so, will update them.

**Review by devendra104 — APPROVED** [review]
ACK

**Pull request merged**